jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -667,8 +672,8 @@ boolean tryAcquire()
                 int usageCount = AtomicBiInteger.getHi(encoded);
                 boolean closed = usageCount < 0;
                 int inUseCount = AtomicBiInteger.getLo(encoded);
-                int maxUsageCount = Pool.this.maxUsageCount;
-                if (closed || !checkInUse.apply(pooled, inUseCount) || (maxUsageCount > 0 && usageCount >= maxUsageCount))
+                int maxUsageCount = Pool.this.maxUsage;
+                if (closed || !available.test(pooled, inUseCount) || (maxUsageCount > 0 && usageCount >= maxUsageCount))
```

Comment:
Ouch! `inUseCount` is going to go through `int -> Integer -> int` transformations just for this call to happen... on the fast path! My guts tell me this change alone is easily going to cost ~20% throughput.

If we only take perf into account, making `available` a simple `Predicate` would be enough as we could delegate the counter to the predicate itself. But I'm not sure it can be done without breaking backward compatibility.